### PR TITLE
refactor: helpers to parse benchmark arguments

### PR DESCRIPTION
**Pull request by coryan**
Remove code duplication between the Storage and Pub/Sub benchmarks. Even
I don't want to cut&paste this code one more time. Did some minor
simplifications, like using `absl::DurationParse` instead of a custom
parser.

Part of the work for #4565

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5333)

### CODE REVIEW

File: google/cloud/testing_util/command_line_parsing.h [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+ *
+ * The size must be small enough for an in-memory buffer.
+ */
+std::size_t ParseBufferSize(std::string const& val);
+
+/// Parse a string as a duration with support for hours (h), minutes (m), or
+/// second (s) suffixes.
+std::chrono::seconds ParseDuration(std::string const& val);
+
+/// Parse a string as a boolean, with a default value if the string is empty.
```
Comment (devbww) [inline google/cloud/testing_util/command_line_parsing.h:54]: I'm not sure "a default value" is the best term for "the not-present value".  Either way, it is also the result if the boolean fails to parse, whether or not the string is empty.

File: google/cloud/testing_util/command_line_parsing.cc [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+  if (!absl::ParseDuration(val, &d)) {
+    internal::ThrowInvalidArgument("invalid duration: " + val);
+  }
+  return absl::ToChronoSeconds(d);
+}
+
+absl::optional<bool> ParseBoolean(std::string const& val) {
+  if (val.empty()) {
+    return {};
+  }
```
Comment (devbww) [inline google/cloud/testing_util/command_line_parsing.cc:64]: I would suggest omitting this special/redundant case.

File: google/cloud/testing_util/command_line_parsing.cc [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+  std::ostringstream os;
+  os.setf(std::ios::fixed);
+  os.precision(1);
+  os << (static_cast<double>(size) / static_cast<double>(resolution)) << name;
+  return os.str();
+}
+
+std::string Basename(std::string const& path) {
+  // With C++17 we would use `std::filesystem::path`, until then do the poor's
+  // person version.
```
Comment (devbww) [inline google/cloud/testing_util/command_line_parsing.cc:102]: s/poor's person/poor person's/

Although, I might suggest "until then use `find_last_of()`".

**Review by devbww — APPROVED** [review]

File: google/cloud/testing_util/command_line_parsing.cc (hunk 3 above)
Comment (coryan) [reply google/cloud/testing_util/command_line_parsing.cc]: Done.

File: google/cloud/testing_util/command_line_parsing.cc (hunk 2 above)
Comment (coryan) [reply google/cloud/testing_util/command_line_parsing.cc]: Done.

**Pull request merged**